service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageLegalCounselControllerSubmittedTest.java

```diff
@@ -0,0 +1,168 @@
+        when(caseRoleLookupService.getCaseSolicitorRolesForCurrentUser(TEST_CASE_ID))
+            .thenReturn(asList(SolicitorRole.CHILDSOLICITORA));
+    }
+
+    @Test
+    void shouldUpdateCaseRoleAndNotifyModifiedLegalCounsellorsWhenSubmittedEndpointIsCalled() {
+        Pair<String, LegalCounsellor> addedLegalCounsellor =
+            buildLegalCounsellorAndMockUserId(organisationService, "1");
+        Pair<String, LegalCounsellor> maintainedLegalCounsellor =
+            buildLegalCounsellorAndMockUserId(organisationService, "2");
+        Pair<String, LegalCounsellor> removedLegalCounsellor =
+            buildLegalCounsellorAndMockUserId(organisationService, "3");
+
+        List<Element<Child>> childrenInPreviousCaseData = testChildren();
+        childrenInPreviousCaseData.get(0).getValue().setLegalCounsellors(asList(
+            element(maintainedLegalCounsellor.getValue()),
+            element(removedLegalCounsellor.getValue())
+        ));
+        CaseDetails previousCaseDetails = buildCaseDetailsWithGivenChildren(childrenInPreviousCaseData);
+
+        ArrayList<Element<Child>> childrenInCurrentCaseData = new ArrayList<>(childrenInPreviousCaseData);
+        childrenInCurrentCaseData.get(0).getValue().setLegalCounsellors(asList(
+            element(addedLegalCounsellor.getValue()),
+            element(maintainedLegalCounsellor.getValue())
+        ));
+        CaseDetails currentCaseDetails = buildCaseDetailsWithGivenChildren(childrenInCurrentCaseData);
+
+        CallbackRequest callbackRequest = CallbackRequest.builder()
+            .caseDetailsBefore(previousCaseDetails)
+            .caseDetails(currentCaseDetails)
+            .build();
+
+        SubmittedCallbackResponse response = postSubmittedEvent(callbackRequest);
+
+        assertThat(response.getConfirmationHeader()).isNull();
+        assertThat(response.getConfirmationBody()).isNull();
+        String childLastName = childrenInCurrentCaseData.get(0).getValue().getParty().getLastName();
+        assertAsyncActionsHappenToAddedLegalCounsellor(addedLegalCounsellor, childLastName);
+        assertAsyncActionsHappenToRemovedLegalCounsellor(removedLegalCounsellor, childLastName);
+    }
+
+    private CaseDetails buildCaseDetailsWithGivenChildren(List<Element<Child>> childrenInPreviousCaseData) {
+        CaseData previousCaseData = CaseData.builder()
+            .children1(childrenInPreviousCaseData)
+            .caseName("testCaseName")
+            .build();
+        return CaseDetails.builder()
+            .id(TEST_CASE_ID_AS_LONG)
+            .data(caseConverter.toMap(previousCaseData))
+            .build();
+    }
+
+    private void assertAsyncActionsHappenToAddedLegalCounsellor(Pair<String, LegalCounsellor> addedLegalCounsellor,
+                                                                String childLastName) {
+        LegalCounsellor legalCounsellor = addedLegalCounsellor.getValue();
+        await().untilAsserted(() -> {
```

Comment:
I think we use a method called checkUntil to verify async assertions

Reply:
It probably uses this under the hood

Reply:
Cool Will change.